FsAutoComplete/CompilerServiceInterface.fs

```diff
@@ -0,0 +1,159 @@
+namespace FsAutoComplete
+
+open System
+open System.IO
+open Microsoft.FSharp.Compiler
+open Microsoft.FSharp.Compiler.SourceCodeServices
+
+type ParseAndCheckResults(parse: FSharpParseFileResults,
```

Comment:
Most of the methods in this type are calling `Async.RunSynchronously` - so it makes me wonder if they should be themselves asynchronous and the caller should take care of blocking?

(I guess it does not matter now - but it could be useful for cancellation and it might make this file actually reusable for other projects. I think I could have a use for this in some of the web-based F# editors like the one on www.fun3d.net!)